### PR TITLE
Changed master total PendingQueueSize to PendingQueueSize per task type

### PR DESCRIPTION
**Pull request by meetshah777**
**Master Pending Queue size per task type**

Earlier we were publishing Total number of pending task. For RCA analysis Pending Queue size per task type will give better understanding. Changed total Queue Size to Queue size per task type

_Tests:_

Added some manual entry. Below are the output of manual entry

_1. Table of Master_PendingQueueSize_
```
sqlite> select * from Master_PendingQueueSize;
create-index|5.0|5.0|5.0|5.0
delete-index|3.0|3.0|3.0|3.0
shard-started|10.0|10.0|10.0|10.0
```
_2. Schema of PendingQueueSize_
```
sqlite> .schema Master_PendingQueueSize
CREATE TABLE Master_PendingQueueSize(Master_PendingTaskType varchar null, sum double null, avg double null, min double null, max double null);
```
 
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/test/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/reader/ReaderMetricsProcessorTests.java [hunk 1]
```diff
@@ -145,20 +146,20 @@ public boolean accept(final File dir, final String name) {
         new MemoryDBSnapshot(conn, MetricName.MASTER_PENDING, 6001L);
     long lastUpdateTime1 = 2000L;
     masterPendingSnap1.setLastUpdatedTime(lastUpdateTime1);
-    Object[][] values1 = {{0}};
+        Object[][] values1 = {{"delete-index", 0}};
```
Comment (yu-sun-77) [inline src/test/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/reader/ReaderMetricsProcessorTests.java:149]: nit: can remove the indention?

File: src/test/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/reader/ReaderMetricsProcessorTests.java [hunk 2]
```diff
@@ -222,11 +223,18 @@ public void testAlignNodeMetrics() throws Exception {
             masterPendingFinal);
 
     Result<Record> res = alignedWindow.fetchAll();
-    assertTrue(1 == res.size());
+        assertTrue(2 == res.size());
```
Comment (yu-sun-77) [inline src/test/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/reader/ReaderMetricsProcessorTests.java:226]: same as above, can we remove extra Tab in this file?

File: src/test/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/reader/ReaderMetricsProcessorTests.java [hunk 3]
```diff
@@ -252,13 +260,15 @@ public void testEmitNodeMetrics() throws Exception {
 
     Result<Record> res = db.queryMetric(MasterPendingValue.MASTER_PENDING_QUEUE_SIZE.toString());
 
-    assertTrue(1 == res.size());
+    assertTrue(2 == res.size());
 
     Record row0 = res.get(0);
-    for (int i = 0; i < row0.size(); i++) {
+        for (int i = 1; i < row0.size(); i++) {
```
Comment (yu-sun-77) [inline src/test/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/reader/ReaderMetricsProcessorTests.java:266]: Can we check for res.get(1) as well?

File: src/test/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/reader/ReaderMetricsProcessorTests.java (hunk 1 above)
Comment (jotok) [reply src/test/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/reader/ReaderMetricsProcessorTests.java:149]: Please run `./gradlew spotlessApply` to format the code. That should solve the indentation issues.

File: src/test/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/reader/ReaderMetricsProcessorTests.java [hunk 4]
```diff
@@ -222,11 +223,18 @@ public void testAlignNodeMetrics() throws Exception {
...
     Result<Record> res = alignedWindow.fetchAll();
-    assertTrue(1 == res.size());
+        assertTrue(2 == res.size());
     Field<Double> valueField =
         DSL.field(MasterPendingValue.MASTER_PENDING_QUEUE_SIZE.toString(), Double.class);
+        Field<String> dimensionField =
+                DSL.field(MasterPendingTaskDimension.MASTER_PENDING_TASK_TYPE.toString(), String.class);
     Double pending = Double.parseDouble(res.get(0).get(valueField).toString());
-    assertEquals(2.2d, pending, 0.001);
+        assertEquals(1.0d, pending, 0.001);
```
Comment (yu-sun-77) [inline src/test/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/reader/ReaderMetricsProcessorTests.java:232]: why this change from 2.2d to 1.0d?

Comment (meetshah777) [reply src/test/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/reader/ReaderMetricsProcessorTests.java:232]: Earlier we have only PendingQueue size and we have 3 windows in test case. It will calculate the three window values and result was 2.2. Now we added tasktype so table looks like
```
-------------------------------------------------------
|Master_PendingTaskType | Master_PendingQueueSize |
-------------------------------------------------------
| create-index          | 1.0                     |
| updateSnapshot       | 3.0                      |
--------------------------------------------------------
```

File: src/test/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/reader/ReaderMetricsProcessorTests.java (hunk 3 above)
Comment (meetshah777) [reply src/test/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/reader/ReaderMetricsProcessorTests.java:266]: Yes. Will add that

**Review by yu-sun-77 — APPROVED** [review]

**Review by khushbr — APPROVED** [review]
